pkg/httpc/client.go

```diff
@@ -113,22 +109,27 @@ func (c *Client) Send(
 		return nil, e
 	}
 
-	if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
+	var data []byte
+	if !c.isRawBody {
+		data, err = ioutil.ReadAll(resp.Body)
 		defer resp.Body.Close()
-		data, _ := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			e := errType.Wrap(err, "Failed to read %s API response", errOriginComponent)
+			log.Warn("SendRequest failed", zap.String("uri", uri), zap.Error(err))
+			return nil, e
```

Comment:
When status code is not ok, it is possible that you cannot read the body I think.

Reply:
```
	// The http Client and Transport guarantee that Body is always
	// non-nil, even on responses without a body or responses with
	// a zero-length body. It is the caller's responsibility to
	// close Body. The default HTTP client's Transport may not
	// reuse HTTP/1.x "keep-alive" TCP connections if the Body is
	// not read to completion and closed.
```

I found some comments about Response.Body, and we've read the body when the status code is not ok before.